pkg/reconciler/revision/resources/deploy.go

```diff
@@ -112,6 +134,20 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 		}
 	}
 
+	// If concurrencyStateEndpoint is enabled, add the serviceAccountToken to QP via a projected volume
+	if cfg.Deployment.ConcurrencyStateEndpoint != "" {
+		podSpec.Volumes = append(podSpec.Volumes, varProjectedVolume)
+
+		for i, container := range podSpec.Containers {
+			if container.Name == QueueContainerName {
+				varProjectedMount := varProjectedVolumeMount.DeepCopy()
```

Comment:
I don't think this `DeepCopy` is necessary here. It is above because the actual spec is changed. In this case, it isn't, I think.